utils/conversions.go

```diff
@@ -198,6 +198,9 @@ func ToJSON(env Environment, val interface{}) (JSONFragment, error) {
 		return JSONFragment(bytes), err
 	}
 
+	// if value is a variable resolver, then use it's default
+	val = ResolveDefault(val)
```

Comment:
By having this come first - we ensure that we always stringify the default

Reply:
Is that right? If someone defines a `String` shouldn't that take precedence over `Default`? (do we have examples when we want those to be different?)

Reply:
I think so - tho it does have the odd effect of meaning `String()` is never called on context items with a `Default()` that returns something other than themselves.

`@contact.field.age` defaults to `@contact.field.age.value`, so
`@(contact.field.age == 30)` should resolve `contact.field.age` as a decimal, i.e. `Default()` then `ToDecimal`
`@(CONCAT(@contact.field.age, "years"))` should resolve `contact.field.age` as a string, and should be consistent with `@(CONCAT(@contact.field.age.value, "years"))`

Resolving the default before converting to a primitive means we don't have two different `String` methods that need to be kept consistent.